lib/index.js

```diff
@@ -47,8 +49,190 @@ api.validateConfiguration = (validatorConfig, callback) =>
   validate('ledger-validator-multiproof', validatorConfig, callback);
 
 api.validateEvent = (event, validatorConfig, callback) => {
-  // determine which proof types should be checked
-
-  // FIXME: Implement
-  callback();
+  async.auto({
+    schema: callback => callback(),
+    signatures: ['schema', (results, callback) =>
+      _validateSignatures(event, callback)]
+  }, err => callback(err));
 };
+
+function _validateSignatures(event, callback) {
+
+  // FIXME: could these parameters just be in the ledger configuration instead
+  // of inventing `ParameterAlgorithm`?
+  const equihashParameterAlgorithm = {
+    VeresOne2017: {n: 64, k: 3}
+  };
+  const signatureType = ['LinkedDataSignature2015', 'EquihashProof2017'];
+
+  // FIXME: authorizationCapability[0].permission is `UpdateDidDocument`, how
+  // does this validator know that the current operation is UpdateDidDocument?
+
+  const proofType = event.input[0].authorizationCapability[0]
+    .permittedProofType.map(type => type.proofType);
+
+  if(!proofType.every(type => signatureType.includes(type))) {
+    return callback(new BedrockError(
+      'Unsupported signature type.',
+      'NotSupportedError', {
+        httpStatusCode: 400,
+        public: true,
+        event,
+      }));
+  }
+
+  const signature = {};
+  proofType.forEach(type => signature[type] = event.signature
+    .filter(s => s.type === type));
+  const hasOneOfEach = proofType.every(type => signature[type].length !== 0);
+
+  // FIXME: the `minimumProofsRequired` property defined now seems inadequate to
+  // indicate that we actually want x number of one sig type, and y number of
+  // another sig type.
+  // For this first use case, we will not accept *only* 2 ld-signatures
+  // or *only* 2 equihash sigs.  `minimumProofsRequired` is currently being
+  // ignored
```

Comment:
Maybe it should be "minimumProofTypesRequired". @msporny?

Reply:
wait for input from @msporny on this one.

Reply:
Yes, let's do "minimumProofTypesRequired"

Reply:
Solution: drop this property all together.  Done.